cmd/kubeapps-apis/plugins/fluxv2/packages/v1alpha1/harbor_api_v2_repo_lister.go

```diff
@@ -0,0 +1,298 @@
+		if len(onePage) < pageSize {
+			more = false
+		}
+	}
+	return repos, nil
+}
+
+// ref https://demo.goharbor.io/#/ping/getPing
+func pingHarbor(ctx context.Context, ref orasregistryv2.Reference, cred orasregistryauthv2.Credential) (string, error) {
+	log.Infof("+pingHarbor")
+	url := fmt.Sprintf("%s/ping", buildHarborRegistryBaseURL(false, ref))
+	resp, err := harborDoHttpRequest(ctx, http.MethodGet, cred, url)
+	if err != nil {
+		return "", err
+	}
+	contentType := resp.Header.Get("Content-Type")
+	if !strings.HasPrefix(contentType, "text/plain") {
+		return "", fmt.Errorf("unexpected response content type: [%s]", contentType)
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		lr := io.LimitReader(resp.Body, 100)
+		if pong, err := ioutil.ReadAll(lr); err != nil {
+			return "", err
+		} else if string(pong) != "Pong" {
+			return "", fmt.Errorf("unexpected response body: %s", string(pong))
+		} else {
+			return string(pong), nil
+		}
+
+	case http.StatusNotFound:
+		return "", errdef.ErrNotFound
+
+	default:
+		err = parseHarborErrorResponse(resp)
+		return err.Error(), err
+	}
+}
+
+//  https://demo.goharbor.io/#/project/headProject
+func harborProjectExists(ctx context.Context, projectName string, ref orasregistryv2.Reference, cred orasregistryauthv2.Credential) error {
+	log.Infof("+projectExists(%s)", projectName)
+	url := fmt.Sprintf("%s/projects?project_name=%s",
+		buildHarborRegistryBaseURL(false, ref), projectName)
+	resp, err := harborDoHttpRequest(ctx, http.MethodHead, cred, url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return nil
+
+	case http.StatusNotFound:
+		return errdef.ErrNotFound
+
+	case http.StatusForbidden:
+		return errors.New("forbidden")
+
+	default:
+		err = parseHarborErrorResponse(resp)
+		return err
+	}
+}
+
+// https://demo.goharbor.io/#/repository/listRepositories
+func listHarborProjectRepositories(ctx context.Context, projectName string, ref orasregistryv2.Reference, cred orasregistryauthv2.Credential, page, pageSize int) ([]string, error) {
+	log.Infof("+listProjectRepositories(%s)", projectName)
+	url := fmt.Sprintf("%s/projects/%s/repositories?page=%d&page_size=%d",
+		buildHarborRegistryBaseURL(false, ref), projectName, page, pageSize)
+	resp, err := harborDoHttpRequest(ctx, http.MethodGet, cred, url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	switch resp.StatusCode {
+	case http.StatusOK:
+		return parseHarborProjectRepositoriesResponse(resp)
+
+	case http.StatusNotFound:
+		return nil, errdef.ErrNotFound
+
+	default:
+		err = parseHarborErrorResponse(resp)
+		return nil, err
+	}
+}
+
+// buildHarborRegistryBaseURL builds the URL for accessing the base API.
+func buildHarborRegistryBaseURL(plainHTTP bool, ref orasregistryv2.Reference) string {
+	scheme := "https"
+	if plainHTTP {
+		scheme = "http"
+	}
+	return fmt.Sprintf("%s://%s/api/v2.0", scheme, ref.Host())
```

Comment:
Just thinking:  would be a good idea to extract all the templates into `const` being defined at the bottom? This way we would have all the paths in the same place (in case the change their API.
Anyway, not really important, as I don't foresee that breaking changes for the major (v2) this lister is intended to.

Reply:
yeah, I don't see them changing v2 API in a backwards incompatible way anywhere soon. Will keep your comment in mind for the next PR(s)